### PR TITLE
Feature/les 430 upload/delete thumbnail for lesson

### PR DESCRIPTION
**Pull request by OxygenO22**

### CODE REVIEW

File: src/components/Thumbnail.tsx [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+
+  return (
+    <div className="thumbnail__wrapper">
+      <div className="thumbnail__left">
+        <div className="thumbnail__left-inner">
+          <div className="thumbnail__picture">
+            <img
+              className="thumbnail__img"
+              src={
+                imageURL
```
Comment (Terapool) [inline src/components/Thumbnail.tsx:58]: Please, separate const for this src before return - too many lines, difficult to read such JSX

const srcUrl = imageURL ? ...

File: src/components/Thumbnail.scss [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+  width: 20px;
+  height: 20px;
+  background: url(./icons/changeFile.svg) no-repeat center;
+  background-size: 15px 17px;
+  margin-right: 6px;
+}
+
+.button__notsee {
+  display: none;
+}
```
Comment (Terapool) [inline src/components/Thumbnail.scss:97]: Please, add empty line at the end of css, ts, tsx, js, jsx files
With an empty line it is easear to add new code

File: src/components/Thumbnail.tsx [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+  };
+  const handleOnChange = (e: ChangeEvent<any>) => {
+    e.preventDefault();
+    console.log("change", e.target.files[0]);
+    if (e.target.files[0].size <= 2000000) {
+      const file = e.target.files[0];
+      setImage(file);
+      fileReader.readAsDataURL(file);
+
+      /* const formData = new FormData();
```
Comment (Terapool) [inline src/components/Thumbnail.tsx:37]: If this code is commented I think we do not need it
Please, remove it as we will spoil our code
99% that we well not uncomment it later after you merge this PR

File: src/components/Thumbnail.tsx [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+    formData.append("file", file)
+    request("https://lessoner.s3.amazonaws.com/", "PUT", formData); */
+    } else {
+      alert(
+        "size to big: " +
+          (Math.floor(e.target.files[0].size) / 1000000).toFixed(2)
```
Comment (Terapool) [inline src/components/Thumbnail.tsx:43]: When you write long numbers use underscore separators:
1000000 -> 1_000_000

This is for better eading of your code
https://v8.dev/features/numeric-separators

Are we really use alert in our interface? Do we have such requirements?

File: src/components/Thumbnail.tsx [hunk 5]
```diff
@@ -0,0 +1,96 @@
...
+          </div>
+          <div className="thumbnail__item-info">
+            <p
+              className="thumbnail__item-name"
+              title={image ? image.name : "Some.png"}
+            >
+              {image ? image.name : "Some image name.png"}
+            </p>
+            <p className="thumbnail__item-size">
+              {image ? (Math.floor(image.size) / 1000000).toFixed(2) : "0.0"} MB
```
Comment (Terapool) [inline src/components/Thumbnail.tsx:75]: Please, separators in 
1_000_000 number

File: src/components/types/types.tsx [hunk 6]
```diff
@@ -1,11 +1,16 @@
 export interface ILessonBack {
   author_id: number;
   category_id: number;
+  created_at: string;
   description: string;
   id: number;
+  image_link?: string;
```
Comment (Terapool) [inline src/components/types/types.tsx:7]: Please, not required fields (with question mark) write at the end, in first place required:

author_id: ...
category_id: ...
....
image_link?: ...

**Review by Terapool — CHANGES_REQUESTED** [review]

**Review by Terapool — APPROVED** [review]

File: src/components/Thumbnail.tsx (hunk 5 above)
Comment (OxygenO22) [reply src/components/Thumbnail.tsx]: done

File: src/components/Thumbnail.tsx (hunk 3 above)
Comment (OxygenO22) [reply src/components/Thumbnail.tsx]: done

File: src/components/Thumbnail.scss (hunk 2 above)
Comment (OxygenO22) [reply src/components/Thumbnail.scss:127]: done

File: src/components/Thumbnail.tsx (hunk 1 above)
Comment (OxygenO22) [reply src/components/Thumbnail.tsx]: done

File: src/components/Thumbnail.tsx (hunk 4 above)
Comment (OxygenO22) [reply src/components/Thumbnail.tsx]: done

File: src/components/types/types.tsx (hunk 6 above)
Comment (OxygenO22) [reply src/components/types/types.tsx]: done

**Review by valer-sky — APPROVED** [review]

**Review by DimitarShemet — APPROVED** [review]

**Pull request merged**